File: src/MoManI.Api/Models/ModelResult.cs
using System;
using System.Collections.Generic;

namespace MoManI.Api.Models
{
    public class ModelResult
    {
        public Guid ModelId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public IEnumerable<VariableResult> VariableResults { get; set; } 
    }
}

File: src/MoManI.Ui/scripts/models/modelResult.ts
import _ = require('lodash');

export class ModelResult {
    id: string;
    name: string;
    description: string;
    variables: IVariable[];

    constructor(modelResult: IModelResult, variables: IVariable[]) {
        this.id = modelResult.modelId;
        this.name = modelResult.name;
        this.description = modelResult.description;
        this.variables = _.map(modelResult.variableResults, v => {
            return _.find(variables, 'id', v.variableId);
        });
    }
}